protobuf/syft_proto/execution/v1/type_wrapper.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package syft_proto.execution.v1;
+option java_package = "org.openmined.syftproto.execution.v1";
+
+message type {
+    /*
+        A type contains the serialization id and string bytes.
+    */
+    int32 id = 1;
+    bytes type = 2;
+}
+
+message NestedTypeWrapper {
+    /*
+    In a nested type we can have 4 scenarios:
+        1. A type.
+        2. A list of types.
+        3. A tuple of types.
+        4. A dict of types.
+    As fields are all optionals in proto3, only one field  from this message should exist.
```

Comment:
Since only one of these fields should exist, this message should probably use a `oneof` to make them explicitly mutually exclusive.

Reply:
You can add fields of any type, but cannot use repeated fields in a oneof construction[0]. 

The workaround would be to make new messages for each repeated type in the oneof construction. Should I go for that?

[0]: https://developers.google.com/protocol-buffers/docs/proto3#using-oneof

Reply:
We did something like this for variable type arguments with the [`Arg`](https://github.com/OpenMined/syft-proto/blob/master/protobuf/syft_proto/types/syft/v1/arg.proto) message, which is used in the `Action` types (e.g. [`ComputationAction`](https://github.com/OpenMined/syft-proto/blob/master/protobuf/syft_proto/execution/v1/computation_action.proto#L20-L21).) Would something like that work?

Reply:
Should be fine now.